src/core/processing/qgsprocessingparameters.h

```diff
@@ -1849,6 +1849,58 @@ class CORE_EXPORT QgsProcessingParameterFolderDestination : public QgsProcessing
 
 };
 
+/**
+ * \class QgsProcessingParameterBand
+ * \ingroup core
+ * A raster band parameter for Processing algorithms.
+  * \since QGIS 3.0
+ */
+class CORE_EXPORT QgsProcessingParameterBand : public QgsProcessingParameterDefinition
+{
+  public:
+
+    /**
+     * Constructor for QgsProcessingParameterBand.
+     */
+    QgsProcessingParameterBand( const QString &name, const QString &description = QString(), const QVariant &defaultValue = QVariant(),
+                                const QString &parentLayerParameterName = QString(),
```

Comment:
Hm - my fault, but this just made me realise that there's an inconsistency here. We should either rename this argument to `parentLayerParameter` or rename the parentLayerParameter() getter to parentLayerParameterName(). Which do you prefer?